### DISCUSSION TITLE
What's the exact point of quickdraw achievement

### DISCUSSION THREAD

**Discussion by rilysh** [community/community#20485] [Other Feature Feedback, Questions, & Ideas]
As the title, what is the point of this badge? I had closed one of my own PR as I mistakenly created two in one repo. However, today I noticed that I got a badge called quickdraw. But why so? I mean, I closed a PR because I mistakenly created two same PR.

I didn't understand, and to be honest, it looks quite unrelated and amateurish badge. ...how this gonna help me when it's totally meaningless achievement.

**Comment by radar-blip** [comment]
Its because you closed it within 5 minutes of opening.
Here is some more info:
https://github.com/Schweinepriester/github-profile-achievements#achievements

**Reply by rilysh** [reply]
Already saw, but you missed the point. Why this achievement or badge even exist? Isn't this will grow up useless PR to get the badge?

**Reply by mejdi14** [reply]
@kiwimoe you're right, this is total nonsense

**Reply by safesploit** [reply]
@mejdi14 I was thinking this myself! Let the _spamming_ begin... Oh, wait I meant 'QUICKDRAW!!'

**Reply by radar-blip** [reply]
lol it makes some people feel like they accomplished something..

**Reply by Tostisto** [reply]
In my opinion, this achievement is completely unnecessary.

**Reply by aexvir** [reply]
all these achievements are pure cringe

**Reply by LarominitaputitA** [reply]
Holi

**Reply by jadehamel** [reply]
Thx!!

**Reply by slancerk** [reply]
i was surprised to have it ,damn looks awesome btw😂

**Reply by surya-murugan-ai** [reply]
this is  the first achievement i have got and it seems kind a cool!

**Reply by adityavijay21** [reply]
can you please tell me how to take that badge?

**Reply by rphair** [reply]
@kiriotheo

1. find some public repository where you don't mind [looking dumb](https://github.com/orgs/community/discussions/20485#discussioncomment-3162898)
2. open up a pull request against its content
3. close that pull request in less than 5 minutes

**Reply by cookierw** [reply]
> @kiriotheo
> 
> 1. find some public repository where you don't mind [looking dumb](https://github.com/orgs/community/discussions/20485#discussioncomment-3162898)
> 2. open up a pull request against its content
> 3. close that pull request in less than 5 minutes

*cringe* please don't do this.

**Reply by rphair** [reply]
@sean-rw it sounds like you completely missed the irony / sarcasm in my posting (even with the explicit "looking dumb" statement). 😝

**Comment by Zoltus** [comment]
They are trying to get people more active.

**Reply by its-alikhokher** [reply]
i really agree with you point,

**Comment by ttrreevvoorr** [comment]
you shot your own PR dead. you are now the fastest gunslinger in the west. that is something you will have to live with.

**Reply by LarominitaputitA** [reply]
Jjajaj

**Reply by CarlosLeonCode** [reply]
LOL

**Reply by tosmartak** [reply]
🤣🤣🤣🤣

**Comment by Mehul2203** [comment]
Hey @kiwimoe, 👋
I would like to answer this question 😊. 

I think GitHub created this achievement just to show how quickly we respond to any PR. 🚀
They are trying to get all of the GitHub users to get more active and to keep us motivated to work with team dynamics in an effective manner. 👍

I hope this answers your question and makes sense. 
Thank you!!

**Comment by santosh2702** [comment]
Totally Agreed with @Mehul2203  Its just for motivation.
Thank You!

**Comment by rphair** [comment]
This badge can made the bearer look a little stupid.  For instance you might open a PR in error and then have to close it yourself because something went wrong about the submission.  Since this badge links to that PR, it exposes the mistake right on the developer's front page.

The problem could be mitigated by awarding it only if the PR "closed within 5 minutes of opening" were a merged PR rather than just a cancelling close.  Otherwise this "gunslinger" icon will frequently be awarding people for shooting themselves in the foot 🤠

**Reply by rromulos** [reply]
I fully agree with you

**Reply by rphair** [reply]
Is there a thread requesting the ability to "hide" an achievement if it links to something embarrassing (or for any reason at all: since some people _might not want_ certain badges)?  I just searched the Profile category & couldn't find such a request... surprising since these pathological cases would really have begun to accumulate by now.

I wouldn't mind "hiding" this badge and then un-hiding it if it ever applies to a more recent PR or issue close that satisfies the badge's original intention... i.e. one that makes me look highly responsive rather than just error-prone. 🤪

**Reply by jovian-explorer** [reply]
https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/managing-contribution-settings-on-your-profile/showing-your-private-contributions-and-achievements-on-your-profile#changing-the-visibility-of-achievements 


This link should help

**Reply by rphair** [reply]
I appreciate the response but ticking that box hides _all_ the achievements, including the legitimate-looking ones. 😞

I would be tempted to submit a GitHub issue requesting that achievements be **individually selectable** in this part of the Settings... if not for most users in this Discussion being happy about these trivial achievements based on their social aspect & unwilling to look at any downside or loss of professionalism. 🤕

**Reply by Carlos-Simim** [reply]
You could make the repository private

**Reply by rphair** [reply]
@Carlos-Simim imagine (as in my case) you opened up some stupid looking PR through a fork (which cannot be made private) on a high visibility repository (which *really* cannot be made private) and then had to close it immediately.

Under most conditions the error would recede into the background along with the closed PRs for the main repository... but not with the friendly Quickdraw icon who will always be there on your profile to say HOWDY DOODY and keep announcing that improperly posted PR to the world.

Of course this is precisely what happened to me, so you can currently see exactly what I'm talking about on my own GitHub profile.  Now just imagine it happening to _you_ and whether _you_ would want to "fix" the problem by hiding the repository itself rather than this little turd of an icon. 🤪

**Reply by Carlos-Simim** [reply]
My bad, I said you could make it private as a possible solution to your problem, I didn't check your profile.
I also think they could change this, however, making one mistake public isn't a big problem in my opinion either, it happens 🤷‍♂️

**Reply by sammyhori** [reply]
> I appreciate the response but ticking that box hides _all_ the achievements, including the legitimate-looking ones. 😞
> 
> I would be tempted to submit a GitHub issue requesting that achievements be **individually selectable** in this part of the Settings... if not for most users in this Discussion being happy about these trivial achievements based on their social aspect & unwilling to look at any downside or loss of professionalism. 🤕

This is now possible, you can now individually select and hide achievements from your profile page

**Comment by fredrikaverpil** [comment]
This is not necessarily a pointless or nonsensical achievement. Let's say you are observant of opened issues and are quick with answering (thus having either the OP or yourself closing it).

But yeah, sure, one could always argue that this is a pointless achievement (or even [making you look stupid](https://github.com/github-community/community/discussions/20485#discussioncomment-3162898)). But I'm sure you can find such angles for most of the other achievements too in that case.

**Reply by rilysh** [reply]
> one could always argue that this is a pointless achievement

It's pointless, no matter what type of statement you want to push here. Everyone have their own opinions, instead of writing "making you look stupid" you can write your own opinion about it. This isn't anything stupid nor embarrassing to argue about it. Furthermore, I found that, this achievement doesn't make a sense just for closing a PR because in any case it didn't fit for the repository.

>  But I'm sure you can find such angles for most of the other achievements too in that case

Yeah, I did find many opinions about other achievements. If you disliked something, can't you talk about it? And here in case, I had not directly write that to remove the whole achievement structure, but did write about "pointless" ones. Such as "YOLO" (which can be different meaning for each person, as it says "you only live once"), "quickdraw" (I may close my PR for any reason or mistakenly create which may not be fit for the repo even completing in limited time) and for these reasons having such colorful icons/achievements on profile feels embarrassing.

**Comment by jovian-explorer** [comment]
It is an achievement that is given if you close a PR or an issue within 5 minutes of opening it. Although it does sound crappy, the original way it was intended to be achieved was if you close someone else's issue or PR, thereby helping them, all in 5 minutes. I know, this may seem trivial, but to be honest, it is a nice achievement to have.

**Comment by TrungJamin** [comment]
<3

**Comment by kycasdzxc** [comment]
this is just fun!🌟

**Comment by tzztson** [comment]
> As the title, what is the point of this badge? I had closed one of my own PR as I mistakenly created two in one repo. However, today I noticed that I got a badge called quickdraw. But why so? I mean, I closed a PR because I mistakenly created two same PR.
> 
> I didn't understand, and to be honest, it looks quite unrelated and amateurish badge. ...how this gonna help me when it's totally meaningless achievement.


###########################################################################################
###########################################################################################

**Comment by tonguyenducmanh** [comment]
Seem like when you delete some repo within 5 minutes created, you achive it

**Comment by luan0ap** [comment]
The same happened to me, I earned this badge by a mistake. Its just fun

**Comment by tzztson** [comment]
If anybody wants to get Galaxy Brain badge, please tell me,
I will teach you step by step

**Reply by rilysh** [reply]
GitHub is for code hosting, you shouldn't *volunteer* for such a stupid thing.

**Comment by DeepanshuGoel1122** [comment]
i achieve quicedraw badge, if i delete that repository by which i get this, badge will remain or not ?

**Comment by jfullstackdev** [comment]
latest update, you can now pick what you want to hide, unlike in the past, it was all or nothing

go to Achievements Tab, click the `Achievements` word, it is a link

![image](https://user-images.githubusercontent.com/47092464/204711684-e7c5ebb6-57f9-4fbf-b8fe-96c2182f41bb.png)

or this URL, it's same thing 

https://github.com/[username]?tab=achievements

you should be seeing like this 

![image](https://user-images.githubusercontent.com/47092464/204711801-dde96c4d-5133-4201-ad8a-9b23b55ed5d2.png)

then, click any badge, hover left then you should be seeing like this 

![image](https://user-images.githubusercontent.com/47092464/204711940-2c5e8b4d-7580-4145-afc0-268f1739d8de.png)

then boom ... you now have the option to hide those badges that you want to hide

**Reply by rphair** [reply]
This is exactly the change & constructive comment about it that I've been waiting for... thanks 😍

**Reply by LGD-hades0701** [reply]
thanks for your reply.

**Comment by WeWuzNidokangz** [comment]
I think it's likely a net negative that this achievement exists in its current form. Regardless of how one feels about gamification, some of the other achievements such as 'Pull Shark' at least seem to primarily incentivize productive behavior. However, 'Quickdraw' is at best likely to be unlocked through accidents, but conceivably might even encourage an inexperienced user to open a PR on another user's repository, specifically with the intention of immediately closing it in order to unlock the achievement. That's not high-impact disruptive behavior, but it's certainly not a good thing.

IMO if an achievement like this is going to exist, it should probably only be unlocked if the PR is merged rather than closed (and in a more reasonable timeframe, like 24 hours).

**Reply by rphair** [reply]
That's what I've also said above but it seems not to attract any attention except from the "peanut gallery" who find the whole situation entertaining... perhaps out point of view was influential in making these achievements "opt-out" (as I was finally able to do with Quickdraw about a month ago).

**Reply by gianlucag** [reply]
Totally agree. As implemented now, it's a shame-badge ! No wonder people wants to selectively hide it. If people were proud of it, it would be all over the place (who never quick drawn it's own PR ?)

**Comment by adityavijay21** [comment]


**Reply by rphair** [reply]
You are getting Quick Draw confused with one of the other badges perhaps. 🤠

**Reply by rilysh** [reply]
Can you stop giving misinformation?
For sure Quickdraw isn't intended to anything like that you mentioned, rather it's a badge that is given when an issue is closed under 5 minutes prior. This is completely useless as it mentions nothing to do with a contribution but a shameful badge which gets attached on profile.

Remember GitHub is *controlled* by a tech giant not by open-souce folks. Also this is pretty outdated discussion to answer.

**Comment by ff137** [comment]
It is absolutely an achievement to have the audacity to open up a faulty pull request, and then very shortly follow up with some due diligence, being like "oh fuck I missed something", and close it.

**Comment by adityavijay21** [comment]
are these achievements downgrading?

**Comment by gmoniava** [comment]
I also closed a issue which I opened mistakenly and got this badge. But when I reopened the issue, for testing purposes, the badge was still there.

**Comment by ahsanba-id** [comment]
This always seems to happen. I'm a self-learner, I started exploring things and baam! It's not like I want my future clients to see my moron.

**Comment by jorgcastellano** [comment]
Go to Achidvement, Regards

**Comment by theurikarue** [comment]
> As the title, what is the point of this badge? I had closed one of my own PR as I mistakenly created two in one repo. However, today I noticed that I got a badge called quickdraw. But why so? I mean, I closed a PR because I mistakenly created two same PR.
> 
> I didn't understand, and to be honest, it looks quite unrelated and amateurish badge. ...how this gonna help me when it's totally meaningless achievement.

Go to this site it will help https://github.com/Schweinepriester/github-profile-achievements#achievements

**Comment by theurikarue** [comment]
Not everyone finds it easy to interact with Github at first so i think such an achievements acts to motivate somebody to know more and interact more with Github

**Comment by Bohousek007** [comment]
The "Quickdraw" badge you received is likely related to the speed at which you closed the pull request (PR) after it was created. Some platforms, like GitHub, award this badge when a user closes a PR or issue very quickly after it is opened, usually within a few minutes or even seconds.

The idea behind this badge is to acknowledge quick actions, regardless of the reason behind them. In your case, you closed the PR quickly because you realized it was a duplicate. The system recognized the speed of your action and awarded you the "Quickdraw" badge.

So, while the badge might seem odd given the circumstances, it's simply a reflection of how promptly you addressed the situation. It's more about the speed of your response than the context of why the PR was closed.
